src/SFA.DAS.LevyTransferMatching/Application/Commands/SetApplicationAcceptance/SetApplicationAcceptanceCommandHandler.cs

```diff
@@ -22,26 +22,43 @@ public SetApplicationAcceptanceCommandHandler(ILogger<SetApplicationAcceptanceCo
 
         public async Task<bool> Handle(SetApplicationAcceptanceCommand request, CancellationToken cancellationToken)
         {
-            if (request.Acceptance != Types.ApplicationAcceptance.Accept)
+            if (request.Acceptance == Types.ApplicationAcceptance.Accept)
             {
-                throw new NotImplementedException();
-            }
+                _logger.LogInformation($"Accepting funding for Application {request.ApplicationId}. {request}");
 
-            _logger.LogInformation($"Accepting funding for Application {request.ApplicationId}. {request}");
+                var apiRequestData = new AcceptFundingRequestData
+                {
+                    UserId = request.UserId,
+                    UserDisplayName = request.UserDisplayName,
+                    AccountId = request.AccountId,
+                    ApplicationId = request.ApplicationId
+                };
 
-            var apiRequestData = new AcceptFundingRequestData
-            {
-                UserId = request.UserId,
-                UserDisplayName = request.UserDisplayName,
-                AccountId = request.AccountId,
-                ApplicationId = request.ApplicationId
-            };
+                var apiRequest = new AcceptFundingRequest(request.ApplicationId, request.AccountId, apiRequestData);
 
-            var apiRequest = new AcceptFundingRequest(request.ApplicationId, request.AccountId, apiRequestData);
+                var result = await _levyTransferMatchingService.AcceptFunding(apiRequest, cancellationToken);
+                return result.StatusCode == HttpStatusCode.NoContent;
+            }
+            else if (request.Acceptance == Types.ApplicationAcceptance.Withdraw)
+            {
+                _logger.LogInformation($"Withdrawing Application {request.ApplicationId}. {request}");
 
-            var result = await _levyTransferMatchingService.AcceptFunding(apiRequest, cancellationToken);
+                var apiRequestData = new WithdrawApplicationRequestData
+                {
+                    UserId = request.UserId,
+                    UserDisplayName = request.UserDisplayName,
+                    AccountId = request.AccountId,
+                    ApplicationId = request.ApplicationId
+                };
 
-            return result.StatusCode == HttpStatusCode.NoContent;
+                var apiRequest = new WithdrawApplicationRequest(request.ApplicationId, request.AccountId, apiRequestData);
+                await _levyTransferMatchingService.WithdrawApplication(apiRequest, cancellationToken);
+                return true;
```

Comment:
Could we not check the status of the inner API request here, and return a bool based on that (similarly to how this is done with acceptance)?